storage/database/dynamodb.go

```diff
@@ -0,0 +1,479 @@
+package database
+
+import (
+	"bytes"
+	"fmt"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/rcrowley/go-metrics"
+
+	"github.com/pkg/errors"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/dynamodb"
+	"github.com/aws/aws-sdk-go/service/dynamodb/dynamodbattribute"
+	"github.com/klaytn/klaytn/log"
+)
+
+var dataNotFoundErr = errors.New("data is not found with the given key")
+var nilDynamoConfigErr = errors.New("attempt to create DynamoDB with nil configuration")
+
+const dynamoWriteSizeLimit = 400*1024 - 32 // The maximum item size is 400 KB including a key size
+const dynamoBatchSize = 25
+const dynamoMaxRetry = 5
+
+// dynamo table provisioned setting
+const dynamoReadCapacityUnits = 10000
+const dynamoWriteCapacityUnits = 10000
+
+// batch write
+const WorkerNum = 10
+const itemChanSize = WorkerNum * 2
+
+var overSizedDataPrefix = []byte("oversizeditem")
+
+type DynamoDBConfig struct {
+	Region             string
+	Endpoint           string
+	TableName          string
+	ReadCapacityUnits  int64
+	WriteCapacityUnits int64
+}
+
+type batchWriteWorkerInput struct {
+	items []*dynamodb.WriteRequest
+	wg    *sync.WaitGroup
+}
+
+type dynamoDB struct {
+	config *DynamoDBConfig
+	db     *dynamodb.DynamoDB
+	fdb    fileDB
+	logger log.Logger // Contextual logger tracking the database path
+
+	// worker pool
+	quitCh  chan struct{}
+	writeCh chan *batchWriteWorkerInput
+
+	batchWriteTimeMeter       metrics.Meter
+	batchWriteCountMeter      metrics.Meter
+	batchWriteSizeMeter       metrics.Meter
+	batchWriteSecPerItemMeter metrics.Meter
+	batchWriteSecPerByteMeter metrics.Meter
+}
+
+type DynamoData struct {
+	Key []byte `json:"Key" dynamodbav:"Key"`
+	Val []byte `json:"Val" dynamodbav:"Val"`
```

Comment:
```suggestion
	Key []byte `json:"K" dynamodbav:"K"`
	Val []byte `json:"V" dynamodbav:"V"`
```
What do you think of shrinking the size to one? The item size is limited by `dynamoWriteSizeLimit`.

Reply:
I change the code giving 1 KB buffer for the length of names and key. I think it can handle 6 length names.